lib/full_url_for.js

```diff
@@ -19,10 +19,12 @@ function fullUrlForHelper(path = '/') {
     if (/^(\/\/|http(s)?:)/.test(path)) return path;
 
     const sitehost = parse(config.url).hostname || config.url;
-    const data = new URL(path, `http://${sitehost}`);
-
-    // Exit if input is an external link or a data url
-    if (data.hostname !== sitehost || data.origin === 'null') return path;
+    let data = null;
+    try {
+      data = new URL(path, `http://${sitehost}`);
+      // Exit if input is an external link or a data url
+      if (data.hostname !== sitehost || data.origin === 'null') return path;
+    } catch {}
```

Comment:
```suggestion
    let data;
    try {
      data = new URL(path, `http://${sitehost}`);
    } catch {}

    // if there is invalid url, data should be undefined
    if (typeof data !== 'object') return path;

    // Exit if input is an external link or a data url
    if (data.hostname !== sitehost || data.origin === 'null') return path;
```

- Use `typeof data !== 'object'` as it has consistent behavior, see https://github.com/hexojs/hexo-util/blob/d08cd68582a39cb0f3357f5d038a6b266551cfc9/lib/is_external_link.js#L31-L35
- `try..catch` without error parameter syntax is unstable, please avoid using this.

Reply:
I've refactored the code, please review again.